packages/combobox/src/index.tsx

```diff
@@ -250,6 +250,10 @@ const ComboboxContext = createNamedContext(
   {} as IComboboxContext
 );
 
+export function useIsComboboxPopoverVisible() {
+  return useContext(ComboboxContext).isVisible;
+}
+
```

Comment:
We could export `ComboboxContext` as a whole, but I'm opting for exporting individual "slices" of context for a few reasons:

- This is the desired way based on the [Reach UI Philosophies][1]
- [Other libraries][2] follow this same pattern
- It allows us to be more conservative with what parts of context we're okay exposing

[1]: https://gist.github.com/ryanflorence/e5c794e6093d16a69fa88d2112a292f7#exposing-state-for-js-styling
[2]: https://github.com/ReactTraining/react-router/blob/b77283cb75efc909c34e48c7cdcb7f50281f4055/packages/react-router/modules/hooks.js#L9

Reply:
Your instinct is correct, IMO. That said, this hook feels a little less than ergonomic. Curious if we can't establish a better pattern for the library as a whole where we expose bits of state folks should reasonably expect. This would be great for pretty much all components with an indexed collection of descendants.

Reply:
Hmm, I'm not sure I follow. Can you help me understand why exposing this slice of context is not ergonomic? Is there a different export you'd be more comfortable with? What do you think is the best actionable step forward?